### PR TITLE
fix/mapper: make Parse recognize customizedProtocol correctly

### LINKED ISSUE #74: bug:  mappers-sdk's Parse can't parse customizedProtocol correctly
When users use mappers-sdk to generate a customized mapper from template, `Parse` will check protocol name: 
```
if instance.PProtocol.Protocol != "Template" {
	continue
}
```
This is checking `protocols.protocol` in deviceProfile.json.
However, the ConfigMap associated with device-instance didn't contain any protocol named "Template". The "Template" appears in `protocols.protocolConfig.protocolName`
eg:
```
"protocols": [{
	"name": "customized-protocol-xxxx",
	"protocol": "customized-protocol",
	"protocolConfig": {
		"protocolName": "Template"
	},
	"protocolCommonConfig": null
}]
```
So, there is a bug, the checking code mentioned above will always exec continue!
There are two ways to fix:
1. modify CRD and device-instance.yaml.
2. modify the checking code.

The second way is easier

### PR DESCRIPTION
**Pull request by jsparter**
modify _template/mapper/configmap/parse.go, validate instance.PProtocol.ProtocolConfigs.ProtocolName

Fixes #74

Signed-off-by: jsparter <1084237763@qq.com>

### CODE REVIEW

File: _template/mapper/configmap/parse.go [hunk 1]
```diff
@@ -23,8 +23,8 @@ import (
 
 	"k8s.io/klog/v2"
 
-	"github.com/kubeedge/mappers-go/mappers/common"
 	"github.com/kubeedge/mappers-go/mappers/Template/globals"
+	"github.com/kubeedge/mappers-go/mappers/common"
```
Comment (sailorvii) [inline _template/mapper/configmap/parse.go:27]: It's better to leave the old for the alphabet order.

File: _template/mapper/configmap/parse.go [hunk 2]
```diff
@@ -57,8 +57,21 @@ func Parse(path string,
...
+		if instance.PProtocol.Protocol != "customized-protocol" {
 			continue
+		} else {
+			cprotocol := struct {
+				ProtocolName string `json:"protocolName"`
+				ConfigData   json.RawMessage
+			}{}
+			err = json.Unmarshal(instance.PProtocol.ProtocolConfigs, &cprotocol)
+			if err != nil {
+				klog.Error("customized-protocol unmarshal error:", err)
```
Comment (sailorvii) [inline _template/mapper/configmap/parse.go:69]: klog.Errorf("customized-protocol unmarshal error: %v", err)

**Review by sailorvii — APPROVED** [review]
Many thanks for this fix.
Some minor comments.

**Comment by sailorvii** [comment]
Refine your commit message referring to other pr.

File: _template/mapper/configmap/parse.go (hunk 1 above)
Comment (jsparter) [reply _template/mapper/configmap/parse.go:27]: Sorry, I didn't aware of it, it's not what I want to do. I'll do it better!

**Pull request merged**